### PR TITLE
Support for Tricore cpu type argument

### PR DESCRIPTION
**Pull request by c-urly**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

Before, Cpu type was default to Rider-B when disassembling an instruction via rasm2
We have the option -c for cpu type that we can use for Tricore variant ISA.
Added support to get the cpu type argument for Tricore arch.

Screenshot 2020-03-07 at 11 38 59 PM

...

**Test plan**

...

**Closing issues**
#15912 

...

### CODE REVIEW

**Comment by radare** [comment]
This 2 in the output looks like a nibble :?

**Comment by c-urly** [comment]
Its Opcode size.

File: libr/asm/p/asm_tricore.c [hunk 1]
```diff
@@ -53,7 +53,24 @@ static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
 	disasm_obj.fprintf_func = &generic_fprintf_func;
 	disasm_obj.stream = stdout;
 
-	disasm_obj.mach = 2; // select CPU TYPE
+	// cpu type
+	if (a->cpu && *a->cpu) {
+		if (!strcmp (a->cpu, "generic")) {
+			disasm_obj.mach = 0x00000000;
```
Comment (pelijah) [inline libr/asm/p/asm_tricore.c]: What about using enum?

Comment (c-urly) [reply libr/asm/p/asm_tricore.c]: Yeah i was thinking that too. But then I was not able to come up with a solution. Do you have  one?

Comment (XVilka) [reply libr/asm/p/asm_tricore.c]: Just create enum:
```c
enum {
   TRICORE_CPU_GENERIC = 0x000000,
   TRICORE_CPU_RIDER_A = 0x000001,
   ...
}
```

Comment (c-urly) [reply libr/asm/p/asm_tricore.c]: Oh you mean when assigning value use enum, i thought to use enum in comparison logic.

Comment (radare) [reply libr/asm/p/asm_tricore.c]: maybe worth moving this switch case into a separate function like disasm_obj.mach = cpu_to_mach (a->cpu);

File: libr/asm/p/asm_tricore.c [hunk 2]
```diff
@@ -53,7 +53,24 @@ static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len) {
...
+			disasm_obj.mach = 0x00000002;
+		} else if (!strcmp (a->cpu, "v2")) {
+			disasm_obj.mach = 0x00000004;
+		} else if (!strcmp (a->cpu, "pcp")) {
+			disasm_obj.mach = 0x00000010;
+		} else if (!strcmp (a->cpu, "pcp2")) {
+			disasm_obj.mach = 0x00000020;
+		}
+	} else {
+			disasm_obj.mach = 0x00000002;
```
Comment (radare) [inline libr/asm/p/asm_tricore.c]: wrong indent

File: libr/asm/p/asm_tricore.c (hunk 1 above)
Comment (c-urly) [reply libr/asm/p/asm_tricore.c]: okay

**Comment by c-urly** [comment]
@radare .

File: libr/asm/p/asm_tricore.c [hunk 3]
```diff
@@ -15,8 +15,36 @@
 static unsigned long Offset = 0;
 static RStrBuf *buf_global = NULL;
 static ut8 bytes[128];
+enum {
+  TRICORE_GENERIC = 0x00000000,
```
Comment (radare) [inline libr/asm/p/asm_tricore.c]: use tabs to indent

File: libr/asm/p/asm_tricore.c [hunk 4]
```diff
@@ -15,8 +15,36 @@
...
+  TRICORE_V2      = 0x00000004,
+  TRICORE_PCP     = 0x00000010,
+  TRICORE_PCP2    = 0x00000020
+};
+
+static int cpu_to_mach(char *cpu_type) {
+	if (cpu_type && *cpu_type) {
+		if (!strcmp (cpu_type, "generic")) {
+			return TRICORE_GENERIC;
+		} else if (!strcmp (cpu_type, "rider-a")) {
```
Comment (radare) [inline libr/asm/p/asm_tricore.c]: no need to else after return

File: libr/asm/p/asm_tricore.c (hunk 3 above)
Comment (c-urly) [reply libr/asm/p/asm_tricore.c]: Okay got it.
This struct i copied it from tricore.h

File: libr/asm/p/asm_tricore.c (hunk 4 above)
Comment (c-urly) [reply libr/asm/p/asm_tricore.c]: I see. My bad.

**Pull request merged**